### ISSUE TITLE
Upgrade fails - fxmlrpc?

### ISSUE THREAD

**Issue by pacija** [AzuraCast/AzuraCast#4213]
**Using Docker installation method?**
No

**AzuraCast version:**
 AzuraCast • v0.12.2 Stable • Ansible • PHP 7.4

**Host Operating System:**
Ubuntu 20.04.1 LTS

**Describe the bug:**
update.sh falls with runtime exception

**Steps to reproduce:**
run sudo ./update.sh

**Expected behavior:**
Azuracast should update to latest stable version.

**Relevant logs:**
```
[RuntimeException]
Failed to execute git checkout '8f053ee53d27da796adff3e1ca929f48e7171c1f'
-- && git reset --hard '8f053ee53d27da796adff3e1ca929f48e7171c1f' --
fatal: reference is not a tree: 8f053ee53d27da796adff3e1ca929f48e7171c1f
```
I noticed above reference is declared in `composer.lock`:

```
        {
            "name": "lstrojny/fxmlrpc",
            "version": "dev-php8",
            "source": {
                "type": "git",
                "url": "https://github.com/VincentLanglet/fxmlrpc.git",
                "reference": "8f053ee53d27da796adff3e1ca929f48e7171c1f"
            },
```

Thank you in advance,

**Comment by SlvrEagle23** [comment]
@pacija This is a known issue that has since been fixed in our Rolling Release version; you may need to switch to the rolling release branch for now until the next stable tagged release. Note that this issue isn't present for our Docker installations as those have dependencies installed at build-time.

**Comment by pacija** [comment]
@SlvrEagle23 Thanks, upgraded to rolling and ran full upgrade, my station is back online:

`Rolling Release #fc3854d (2021-05-26 3:44) • Ansible • PHP 8.0`

Slight off-topic, but as the issue got labeled as "unsupported", here goes.

Azuracast is great, but I think it is really unfortunate that you push docker as the only supported way of using it. It has its own set of problems and it hides too much from the user. Ansible is convenient but it has the same problem of too much automation without I guess quite a number of users not understanding the interaction of components, and where to look at if something goes wrong (eg. is it mysql? or nginx? or php-fpm? or liquidsoap? or icecast? or azuracast (core)?

When you put days into scanning tens of thousands of tracks, creating dozens of playlists and fine-tune programming, it is not easy to "just delete everything and recreate docker containers". Docker is hell to troubleshoot.

I think it would be great if AzuraCast kept separate repo for its core web / php application and focused on its development, without losing too much effort on other components such as databases, web servers, php implementations, liquidsoap, icecast etc.

**Comment by SlvrEagle23** [comment]
@pacija This is a common sentiment from our more technically advanced users, but the biggest problem is that we cater to an audience with an extremely wide range of technical skill and comfort level with Linux. The majority of users don't need (or want) to know about the specifics of how the various pieces of software interconnect, and can (and often do) run into problems if the default instructions require too much technical skill to complete.

Docker is the only established solution for this problem with broad support right now on the server-side. There are other technologies (snap, appimage, etc) for client-side applications, but the server-side is largely relegated to Docker or Kubernetes, and if you think Docker's needlessly complex (which it can be), try Kubernetes! That stuff scares even me!

Our main web application is, in theory, completely usable independent of either our Ansible or Docker installation, but we're constantly changing the software we depend on or the version of the software we're using; for example, we just recently completed the jump to PHP 8.0, and if we hadn't had automated scripts for all users, we couldn't reliably count on everyone being on the latest version of PHP, and our upgrade process would be far harder.

Basically, if you're the type of "power user" who wants to use our PHP web application by itself, you can already do so by installing it via Composer directly; for most of our users, even providing instructions could be a recipe for disaster, so we don't, simply to reduce our support load to something a team as small as ours can handle.
